BlueprintUI/Sources/Layout/GridRow.swift

```diff
@@ -0,0 +1,244 @@
+
+            return CGSize(width: constraint.width.maximum, height: height)
+        }
+
+        func layout(size: CGSize, items: [(traits: Width, content: Measurable)]) -> [LayoutAttributes] {
+            guard items.count > 0 else {
+                return []
+            }
+
+            return _frames(in: SizeConstraint(size), isExactConstraint: true, items: items)
+                .map(LayoutAttributes.init)
+        }
+
+        /// Compute child frames in a given layout space.
+        ///
+        /// First, we measure the size of absolutely-sized  children. Next, we measure the size of
+        /// proportionally-sized children, allocating them a proportion of the layout width available after spacing
+        /// and absolutely-sized children are subtracted.
+        ///
+        /// Finally, we compute the frames of children within the row from left to right, assigning a y-origin
+        /// and height which fulfill the row's vertical alignment.
+        ///
+        /// - Parameters:
+        ///   - constraint: A constraint describing the available layout space.
+        ///   - isExactConstraint: Whether or not the constraint describes the row's size exactly. During measurement,
+        ///   it does not. During layout, it does.
+        ///   - items: Children to layout.
+        private func _frames(
+            in constraint: SizeConstraint,
+            isExactConstraint: Bool = false,
+            items: [(traits: Width, content: Measurable)]
+        ) -> [CGRect] {
+            var sizes: [CGSize] = (0..<items.count).map { _ in .zero }
+
+            // Group children by their sizing. Maintain child order by also storing index.
+            var absolutelySized: [(index: Int, width: CGFloat, content: Measurable)] = []
+            var proportionallySized: [(index: Int, proportion: CGFloat, content: Measurable)] = []
+            items.enumerated().forEach { (index, item) in
+                switch item.traits {
+                case .absolute(let width):
+                    absolutelySized.append((index, width, item.content))
+                case .proportional(let proportion):
+                    proportionallySized.append((index, proportion, item.content))
+                }
+            }
+
+            // Measure absolutely-sized children and record the total width they require.
+            var absoluteItemWidth: CGFloat = 0
+            absolutelySized.forEach { (index, width, content) in
+                absoluteItemWidth += width
+                let constraint = SizeConstraint(width: .atMost(width), height: constraint.height)
+                sizes[index] = CGSize(width: width, height: content.measure(in: constraint).height)
+            }
+
+            // Measure proportionally-sized children, allotting them a portion of the available space.
+            let availableWidth = constraint.width.maximum - (spacing * CGFloat(items.count - 1)) - absoluteItemWidth
```

Comment:
This makes a lot of sense to me, though similarly to what follows below, we want to make sure we account for the `.exactly` case, where it is true that we can use the given value.

Because I was only testing in vertical containers (`ScrollView` and `Column` mostly), I missed the potential flaws in this shortcut.

Reply:
I followed this approach, and ended up with the following:

```
var scale: CGFloat = 0
items.forEach { (_, proportion, content) in
    let width = content.measure(in: constraint).width
    scale = max(scale, width / proportion)
}

items.forEach { (index, proportion, content) in
    let width = scale * proportion
    let constraint = SizeConstraint(width: .atMost(width), height: constraint.height)
    sizes[index] = CGSize(width: width, height: content.measure(in: constraint).height)
}
```

I'll work on making this as legible as possible. Thanks for the guidance.